services/sagemaker/apis/v1alpha1/processing_job.go

```diff
@@ -57,8 +57,8 @@ type ProcessingJobStatus struct {
 // ProcessingJob is the Schema for the ProcessingJobs API
 // +kubebuilder:object:root=true
 // +kubebuilder:subresource:status
-// +kubebuilder:printcolumn:name="FailureReason",type=string,JSONPath=`.status.failureReason`
 // +kubebuilder:printcolumn:name="ProcessingJobStatus",type=string,JSONPath=`.status.processingJobStatus`
+// +kubebuilder:printcolumn:name="FailureReason",type=string,JSONPath=`.status.failureReason`
```

Comment:
in order to prevent this kind of change, we need to sort the slice of `PrinterColumn` struct pointers.

Instead of iterating directly over the `CRD.AdditionalPrinterColumns` field:

https://github.com/aws/aws-controllers-k8s/blob/main/templates/apis/crd.go.tpl#L44

we need to make the AdditionalPrinterColumns field unexported and add an `AdditionalPrinterColumns` exported method of CRD that sorts the slice of `PrinterColumn` structs by the `Name` field of the `PrinterColumn` struct.

/cc @RedbackThomson